src/common/error.c

```diff
@@ -371,9 +373,12 @@ errorInternalThrow(
     errorContext.error.functionName = functionName;
     errorContext.error.fileLine = fileLine;
 
-    // Assign message to the error
-    strncpy(messageBuffer, message, sizeof(messageBuffer));
-    messageBuffer[sizeof(messageBuffer) - 1] = 0;
+    // Assign message to the error. If errorMessage() is passed as the message there is no need to make a copy.
```

Comment:
This isn't just an optimization; it fixes a potential (but likely harmless) bug. With strncpy, the strings are not supposed to overlap.

I've never liked the way strncpy handles EOS.  We could use the Berkeley strlcpy which makes that extra assignment unnecessary

Reply:
True enough, but we've never used it in this way before, so it was never a problem. Normally when you want the same message you also want the same error type so we just RETHROW().

Still, a bug of course, but not a live one as far as I can tell.

Reply:
I amended the commit message to make it clear that a bug was being fixed to prevent confusion in the future.